session_0/word_processing.rb

```diff
@@ -34,11 +48,32 @@ def words_with_prefix(prefix, words)
 # The function `similarity_score` takes two words and returns the
 # similarity score (an integer).
 def similarity_score(word_1, word_2)
-  raise NotImplementedError # TODO
+  commonPrefix=''
+  i=0
+  while i < word_1.size do
+    if word_1[i]==word_2[i]
+      commonPrefix+=word_1[i]
+    else
+      break
+    end
+    i+=1
+  end
+  commonPrefix.size
 end
 
 # Given a chosen word and an array of words, return an array of word(s)
 # with the maximum similarity score in the order they appear.
 def most_similar_words(chosen_word, words)
-  raise NotImplementedError # TODO
-end
+  maxScore=0
+  array=Array.new
+  words.each do |word|
+    score=similarity_score(chosen_word,word)
+    if score==maxScore && maxScore>0
```

Comment:
Is the `maxScore > 0` correct?

Or in other words, what be the answer of this:

```ruby
most_similar_word('b', ['a', 'aa', 'aaa'])
```

Since the similarity score of all words is zero and thus equal to maximum (which is also zero), shouldn't the answer be `['a', 'aa', 'aaa']`?

What does your code return?